File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/model/CardsBucket.java
package com.dbottillo.mtgsearchfree.model;

import com.dbottillo.mtgsearchfree.util.LOG;

import java.util.ArrayList;
import java.util.List;

public class CardsBucket {

    String key;
    List<MTGCard> cards;

    public CardsBucket(MTGSet set, List<MTGCard> cards) {
        LOG.d("creating bucket with key: " + set.getName());
        this.key = set.getName();
        this.cards = cards;
    }

    public CardsBucket(String key, List<MTGCard> cards) {
        LOG.d("creating bucket with key: " + key);
        this.key = key;
        this.cards = cards;
    }

    public List<MTGCard> getCards() {
        LOG.d("cards of bucket " + key + " requested");
        return cards;
    }

    public boolean isValid(String otherKey) {
        LOG.d("current bucket contains " + key + " an key required is: " + otherKey);
        return key.equals(otherKey);
    }

    public void setCards(List<MTGCard> cards) {
        this.cards = cards;
    }

    public String getKey() {
        return key;
    }
}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/resources/CardsBucket.kt
package com.dbottillo.mtgsearchfree.resources

import java.util.*

class CardsBucket {

    var set: MTGSet
    var cards: ArrayList<MTGCard>

    constructor(set: MTGSet, cards: ArrayList<MTGCard>) {
        this.set = set
        this.cards = cards
    }

    constructor(search: String, cards: ArrayList<MTGCard>) {
        this.set = MTGSet(-1)
        this.cards = cards
        this.set.name = search
    }

    fun getKey(): String {
        return set.name
    }

    fun isValid(key: String?): Boolean {
        return true
    }
}
